### ISSUE TITLE
Why you are use custom gradint appiler?

### ISSUE THREAD

**Issue by dm-mch** [miyosuda/async_deep_reinforce#21]
Hello, thank you for your code!
Why you are not use standart apply_gradients for rms optimizer? Any sync issue for multithreading? https://www.tensorflow.org/versions/master/api_docs/python/train/optimizers#Optimizer.apply_gradients

**Comment by miyosuda** [comment]
I created this class because I needed to calculate gradient with local loss and variables, but we need to apply calculated gradients to global variables.
I also needed to share "rms" and "momentum" slot among threads, to do SharedRMSProp.

I chose to create my own class, but it might be possible to use standard RMSPropOptimizer, if we call compute_gradients() with local loss and variables, and call apply_gradients with calculated gradients and global variables.
